hub/tasks/x_event_source.py

```diff
@@ -0,0 +1,164 @@
+import os
+from contextlib import asynccontextmanager
+from functools import partial
+
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+from apscheduler.triggers.interval import IntervalTrigger
+from nearai.shared.client_config import DEFAULT_MODEL
+
+from hub.api.v1.auth import AuthToken
+from hub.api.v1.thread_routes import RunCreateParamsBase, ThreadModel, _create_thread, create_run
+from hub.tasks.twitter_client import get_latest_mentions
+
+RESET_BLOCK_ID_ON_START = True  # reusing setting name for now
+
+# Path to store the last read tweet id
+TWEET_STATUS_FILE = "last_tweet.txt"
+
+scheduler = AsyncIOScheduler()
+
+# todo pull from registry 'integration' item
+EVENT_SOURCE_CONFIG = {"user_id": "1867270324649160704"}
+
+# todo pull from agent metadata
+LISTENER_CONFIG = {"agent_name": "flatirons.near/nearsecretagent/0.0.1"}
```

Comment:
I assume the best dev experience is to set a field `trigger_by_users_mentions`:  `user_id` (or maybe even [`user_id1`, `user_id2`]) in the metadata and here to select all of them from the database

Reply:
Yeah, maybe `triggers: twitter: mentions: user_id`. In any case, event sources and trigger config can go in another PR.